### PR TITLE
Use springboot and mybatis connect with postgresql add spring security with encrypt password method and postman scripts

### PR DESCRIPTION
**Pull request by XuXING0430**
1.Use flyway finish the complete data migrate..
2.Use Spring Boot and Mybatis to implementation a JDBC connection with postgresql

### CODE REVIEW

File: src/main/java/com/kingland/bootcamp/utils/FlywayScript.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ */
+package com.kingland.bootcamp.utils;
+
+import org.flywaydb.core.Flyway;
+/**
+ * @author KSC
+ *
+ * Use flyway complete data migrate.
+ */
+public class FlywayScript {
```
Comment (antonyhot) [inline src/main/java/com/kingland/bootcamp/utils/FlywayScript.java]: PR1: Please remove this main method and instead use a spring ApplicationRunner.

File: src/main/resources/db/migration/postgres/V2__Add_accounts.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Insert into accounts(username,password) values ('livepool','18942020');
```
Comment (antonyhot) [inline src/main/resources/db/migration/postgres/V2__Add_accounts.sql]: PR1: You should encrypt your password in database.

**Review by antonyhot — CHANGES_REQUESTED** [review]
@XuXING0430 please check my comments.

File: src/main/java/com/kingland/bootcamp/utils/FlywayScript.java (hunk 1 above)
Comment (XuXING0430) [reply src/main/java/com/kingland/bootcamp/utils/FlywayScript.java]: @antonyhot I delete this file ,flyway in spring boot project do not need to write a  file to complete data migrate.

File: src/main/resources/db/migration/postgres/V2__Add_accounts.sql (hunk 2 above)
Comment (XuXING0430) [reply src/main/resources/db/migration/postgres/V2__Add_accounts.sql]: @antonyhot I try to use ENCODE to encrypt the password.So may use DECODE to decoding the password.

**Comment by XuXING0430** [comment]
sdadasda

**Comment by XuXING0430** [comment]
@antonyhot  This version I try to add spring security in the old version project. This version  achieve simple login features,It can identify three situations:

1.The username is not in the database,the front page return "null:.

2. The username is in the database but the password is wrong,the front page may return "Bad credentials".

3 When the username and password is all correct,the front page may return the account‘s information.

Looking forward to your review comments and suggestions,Thanks.

File: Postman_API_Testing/Scenario_Testing_Cases/Test API - My First Spingboot and Spring security.postman_collection.json [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+		"_postman_id": "7e51e299-06f8-4f3e-be4f-20ddd62cc85d",
+		"name": "Test API - My First Spingboot and Spring security",
+		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json"
+	},
+	"item": [
+		{
+			"name": "Test API - Find all accounts",
+			"event": [
+				{
+					"listen": "test",
```
Comment (vigao) [inline Postman_API_Testing/Scenario_Testing_Cases/Test API - My First Spingboot and Spring security.postman_collection.json]: Please format code

File: src/main/java/com/kingland/bootcamp/security/password/BootcampPasswordEncoder.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ */
+public class BootcampPasswordEncoder implements PasswordEncoder {
+
+    private static BootcampDigestAlgorithm bootcampDigestAlgorithm;
+
+    @Override
+    public String encode(CharSequence rawPassword) {
+
+        String encodedPassword = "";
+//        String salt =  BootcampDigestAlgorithm.getSalt();
```
Comment (vigao) [inline src/main/java/com/kingland/bootcamp/security/password/BootcampPasswordEncoder.java]: Please remove unused code

File: src/main/java/com/kingland/bootcamp/web/controller/Impl/AccountControllerImpl.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    @RequestMapping(value = "/api/addAccount",method = RequestMethod.GET)
+    public void addAccounts(Account account){
+
+        accountService.addAccounts(account);
+    }
+
+    @GetMapping("index")
+    public Object index(Authentication authentication) {
```
Comment (vigao) [inline src/main/java/com/kingland/bootcamp/web/controller/Impl/AccountControllerImpl.java]: Please uniformly define the return value type.

File: src/main/java/com/kingland/bootcamp/web/controller/Impl/BrowserSecurityControllerImpl.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    private RedirectStrategy redirectStrategy = new DefaultRedirectStrategy();
+
+    @GetMapping(SecurityConst.AUTH_REQUIRE)
+    @ResponseStatus(HttpStatus.UNAUTHORIZED)
+    public BaseResponse requireAuthentication(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        SavedRequest savedRequest = requestCache.getRequest(request, response);
+        if (savedRequest != null) {
+            String targetUrl = savedRequest.getRedirectUrl();
+//            log.info();
```
Comment (vigao) [inline src/main/java/com/kingland/bootcamp/web/controller/Impl/BrowserSecurityControllerImpl.java]: Please remove unused code

File: src/main/resources/application.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    - classpath:db/migration
+
+mybatis:
+  type-aliases-package: com.kingland.bootcamp.domain
+  mapper-locations: classpath:mapper/*.xml
+  configuration:
+    map-underscore-to-camel-case: true
+
+server:
+  port: 8888
```
Comment (vigao) [inline src/main/resources/application.yml]: Please add application

File: src/main/resources/application.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    enabled: true
+
+spring:
+  datasource:
+    url: jdbc:postgresql://localhost:5432/bootcamp
+    username: postgres
+    password: pg123456
+
+flyway:
+  enabled: false
```
Comment (vigao) [inline src/main/resources/application.yml]: enable maybe true?

File: src/main/resources/mapper/AccountDao.xml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        from accounts
+        where username = #{username}
+
+    </select>
+
+    <insert id="addAccounts" parameterType="com.kingland.bootcamp.domain.Account" >
+        insert into accounts (username, password) values (#{username}, #{password})
+    </insert>
+
+    <select id="getAllAccounts" resultType="com.kingland.bootcamp.domain.Account">
```
Comment (vigao) [inline src/main/resources/mapper/AccountDao.xml]: resultMap="resultMap"

File: src/test/java/com/kingland/bootcamp/DBPropertiesTest.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * @author KSC
+ *
+ * Test database properties values
+ */
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = DatabaseProperties.class)
+public class DBPropertiesTest {
```
Comment (vigao) [inline src/test/java/com/kingland/bootcamp/DBPropertiesTest.java]: Path error: The file should be placed under test

File: src/test/java/com/kingland/bootcamp/web/controller/BrowserSecurityControllerTest.java [hunk 11]
```diff
@@ -0,0 +1,4 @@
+package com.kingland.bootcamp.web.controller;
+
+public class BrowserSecurityControllerTest {
```
Comment (vigao) [inline src/test/java/com/kingland/bootcamp/web/controller/BrowserSecurityControllerTest.java]: remove unused file

File: src/main/resources/db/migration/V1__Create_accounts_table_And_Insert_data.sql [hunk 12]
```diff
@@ -0,0 +1,10 @@
+DROP TABLE IF EXISTS bootcamp.accounts;
+
+CREATE TABLE accounts  (
```
Comment (vigao) [inline src/main/resources/db/migration/V1__Create_accounts_table_And_Insert_data.sql]: please add " IF NOT EXISTS "

File: src/main/resources/db/migration/V1__Create_accounts_table_And_Insert_data.sql [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+	user_id bigserial PRIMARY KEY,
```
Comment (vigao) [inline src/main/resources/db/migration/V1__Create_accounts_table_And_Insert_data.sql]: id should not null

**Comment by vigao** [comment]
@XuXING0430  Please format the code with corresponding comments on the methods, and some message can be defined in the Constants file.  Please check my comments.

**Review by vigao — APPROVED** [review]
@XuXING0430  Please format the code with corresponding comments on the methods, and some message can be defined in the Constants file. Please check my comments.

**Pull request merged**